exporter/chronicleforwarderexporter/config.go

```diff
@@ -0,0 +1,96 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chronicleforwarderexporter
+
+import (
+	"fmt"
+
+	"github.com/observiq/bindplane-agent/expr"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.uber.org/zap"
+)
+
+const (
+	// ExportTypeSyslog is the syslog export type.
+	ExportTypeSyslog = "syslog"
+
+	// ExportTypeFile is the file export type.
+	ExportTypeFile = "file"
+)
+
+// Config defines configuration for the Chronicle exporter.
+type Config struct {
+	exporterhelper.TimeoutSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	exporterhelper.QueueSettings   `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings   `mapstructure:"retry_on_failure"`
+
+	// ExportType is the type of export to use.
+	ExportType string `mapstructure:"export_type"`
+
+	// Syslog is the configuration for the connection to the Chronicle forwarder.
+	Syslog SyslogConfig `mapstructure:"syslog"`
+
+	// File is the configuration for the connection to the Chronicle forwarder.
+	File File `mapstructure:"file"`
+
+	// RawLogField is the field name that will be used to send raw logs to Chronicle.
+	RawLogField string `mapstructure:"raw_log_field"`
+}
+
+// SyslogConfig defines configuration for the Chronicle forwarder connection.
+type SyslogConfig struct {
+	confignet.NetAddr `mapstructure:",squash"`
+
+	// TLSSetting struct exposes TLS client configuration.
+	TLSSetting *configtls.TLSClientSetting `mapstructure:"tls"`
+}
+
+// File defines configuration for sending to.
+type File struct {
+	// Path is the path to the file to send to Chronicle.
+	Path string `mapstructure:"path"`
+}
+
+// Validate validates the Chronicle exporter configuration.
+func (cfg *Config) Validate() error {
+	if cfg.ExportType != ExportTypeSyslog && cfg.ExportType != ExportTypeFile {
+		return fmt.Errorf("export_type must be either 'syslog' or 'file'")
```

Comment:
There are a few of these but `fmt.Errorf` should only be used if there is string interpolation. If it's just a raw string use `errors.New`

Reply:
yeah, bad habit, updated.